src/main/scala/com/target/data_validator/validator/ColumnSumCheck.scala

```diff
@@ -0,0 +1,118 @@
+package com.target.data_validator.validator
+
+import com.target.data_validator.{ValidatorCheckEvent, ValidatorCounter, ValidatorError, VarSubstitution}
+import io.circe._
+import io.circe.generic.semiauto._
+import org.apache.spark.sql.{DataFrame, Row}
+import org.apache.spark.sql.catalyst.analysis.UnresolvedAttribute
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.catalyst.expressions.aggregate.Sum
+import org.apache.spark.sql.types._
+
+case class ColumnSumCheck(
+  column: String,
+  minValue: Option[Json] = None,
+  maxValue: Option[Json] = None,
+  inclusive: Option[Json] = None
+)
+  extends ColumnBased(column, Sum(UnresolvedAttribute(column)).toAggregateExpression())
+    with MinMaxChecks {
+
+  override def name: String = "columnSumCheck"
+
+  def boundsAreInclusive: Boolean = inclusive.flatMap(_.asBoolean).getOrElse(false)
+
+  private def createTypedLiteral(json: Json)(implicit dataType: DataType): Expression = {
+    ValidatorBase.createLiteral(dataType, json)
+  }
+
+  private def combineExpressions(
+    maxTest: => Option[Expression],
+    minTest: => Option[Expression]): Expression = {
+    (minTest, maxTest) match {
+      case (None, None) =>
+        val msg = "Both min and max tests were None in columnSumCheck. Were minValue and maxValue defined?"
+        logger.error(msg)
+        addEvent(ValidatorError(msg))
+        Literal.FalseLiteral
+      case (None, Some(maxOnly)) => maxOnly
+      case (Some(minOnly), None) => minOnly
+      case (Some(min), Some(max)) => And(min, max)
+    }
+  }
+
+  override def quickCheck(r: Row, count: Long, idx: Int): Boolean = {
+    implicit val dataType: DataType = r.schema(idx).dataType
+    val rawValueForLogging = r.get(idx)
```

Comment:
For this column-based check, we're only looking that the output of the condTest expression, which for columnSumCheck is `Sum(UnresolvedAttribute(column)).toAggregateExpression()`.

Reply:
What I mean is, does data-validator support nested columns (`x.y.z`) instead of just indexing into a single field by an ordinal, in this case `idx`, on a "flat schema".

Reply:
Oh I guess the check config would be responsible for that? all of the check expressions are projected onto a flat schema organized indexes?